tile.cpp

```diff
@@ -1583,8 +1587,36 @@ serial_feature next_feature(decompressor *geoms, std::atomic<long long> *geompos
 			}
 		}
 
-		if (sf.tippecanoe_minzoom == -1 && z < sf.feature_minzoom) {
+		if (sf.tippecanoe_minzoom == -1) {
+			bool keep = false;
+
+			std::string layername = (*layer_unmaps)[sf.segment][sf.layer];
+			auto count = multiplier_state->count.find(layername);
+			if (count == multiplier_state->count.end()) {
+				multiplier_state->count.emplace(layername, 0);
+				count = multiplier_state->count.find(layername);
+				keep = true;  // the first feature in each layer in each tile is always kept
+			}
+
 			sf.dropped = true;
+
+			if (z >= sf.feature_minzoom || keep) {
+				count->second = retain_points_multiplier;
+
+				if (retain_points_multiplier > 1) {
+					serial_val val;
+					val.type = mvt_bool;
+					val.s = "true";
+
+					sf.full_keys.push_back("tippecanoe:retain_points_multiplier_first");
+					sf.full_values.push_back(val);
+				}
+			}
+
+			if (count->second > 0) {
```

Comment:
This is probably not the best place for the comment, but it's where the question occurs to me -- what prevents the `retain_points_multiplier` logic from applying to non-points. I see that the clustering will only happen with points (kind of obviously), but this `sf.dropped` logic seems like it would come before that. I guess one question I have -- is it possible this would run for lines and polygons but just have no effect?

Reply:
The thing that prevents it is that the drop rate doesn't normally apply to polygon and linestring features, only to points, unless you use `--drop-lines` or `--drop-polygons`. Unless you do, polygon and linestring features will always have a `feature_minzoom` of 0.

Reply:
OK, and I think I understand -- before we just wouldn't mark sf.dropped = true in the first place, now we mark it dropped, then by setting count->second to retain_points_multiplier, we then make it un-dropped.

IF retain_points_multiplier were >1 (which is possible by tippecanoe options, even if we're not using it in Felt), then this would add the same metadata for lines/polygons as it does for points.